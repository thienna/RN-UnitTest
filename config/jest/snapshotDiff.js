import snapshotDiff, { toMatchDiffSnapshot } from 'snapshot-diff';

expect.extend({ toMatchDiffSnapshot });
expect.addSnapshotSerializer(snapshotDiff.getSnapshotDiffSerializer());
