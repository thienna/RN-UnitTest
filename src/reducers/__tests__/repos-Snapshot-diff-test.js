import repos from '../repos';
import { initialState } from '../repos';
import { receiveRepos } from '../../actions/api';
import { repos as data } from '../../../config/jest/mockData';

it('compare with toMatchDiffSnapshot', () => {
  expect(
    initialState
  ).toMatchDiffSnapshot(
    repos(initialState, receiveRepos.success(data)),
    { contextLines: 0 }
  );
});

it('LOAD_REPOS_FAILURE with toMatchDiffSnapshot', () => {
  const error = new Error('Look ma! I am an error');
  expect(
    initialState
  ).toMatchDiffSnapshot(
    repos(initialState, receiveRepos.failure(error)),
    {
      aAnnotation: 'Initial state',
      bAnnotation: 'Failure state',
      contextLines: 0,
    }
  );
});