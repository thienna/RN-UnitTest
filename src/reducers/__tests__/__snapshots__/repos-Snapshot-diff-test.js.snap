// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOAD_REPOS_FAILURE with toMatchDiffSnapshot 1`] = `
Snapshot Diff:
- Initial state
+ Failure state

@@ -2,1 +2,1 @@
-   "error": null,
+   "error": [Error: Look ma! I am an error],
`;

exports[`compare with toMatchDiffSnapshot 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -4,1 +4,20 @@
-   "repos": null,
+   "repos": Array [
+     Object {
+       "description": "Useful description goes here.",
+       "id": 1,
+       "name": "Test 1",
+       "stargazers_count": 50000,
+     },
+     Object {
+       "description": "Useful description goes here.",
+       "id": 2,
+       "name": "Test 2",
+       "stargazers_count": 10000,
+     },
+     Object {
+       "description": "Useful description goes here.",
+       "id": 3,
+       "name": "Test 3",
+       "stargazers_count": 499,
+     },
+   ],
`;
